echopop/analysis.py

```diff
@@ -0,0 +1,600 @@
+    transect_data = edit_transect_columns(analysis_dict["transect"], settings_dict)
+
+    # Crop the mesh grid if the kriged data will not be extrapolated
+    if not settings_dict["extrapolate"]:
+        # ---- Compute the cropped mesh
+        mesh_full = crop_mesh(transect_data, mesh_data, settings_dict)
+        if (settings_dict["verbose"]) & (settings_dict["crop_method"] == "convex_hull"):
+            # ---- Print alert
+            print(
+                "Kriging mesh cropped to prevent extrapolation beyond the defined "
+                f"""`mesh_buffer_distance` value ({settings_dict['mesh_buffer_distance']} nmi)."""
+            )
+
+    else:
+        # ---- Else, extract original mesh dataframe
+        mesh_df = mesh_data.copy()
+        # ---- Extract longitude column name
+        mesh_longitude = [col for col in mesh_df.columns if "lon" in col.lower()][0]
+        # ---- Latitude
+        mesh_latitude = [col for col in mesh_df.columns if "lat" in col.lower()][0]
+        # ---- Rename the dataframe
+        mesh_full = mesh_df.copy().rename(
+            columns={f"{mesh_longitude}": "longitude", f"{mesh_latitude}": "latitude"}
+        )
+
+    # Standardize the x- and y-coordinates, if necessary
+    if settings_dict["standardize_coordinates"]:
+        # ---- Transform transect data geometry (generate standardized x- and y-coordinates)
+        transect_data, d_x, d_y = transform_geometry(transect_data, isobath_data, settings_dict)
+        # ---- Transform mesh grid geometry (generate standardized x- and y-coordinates)
+        mesh_full, _, _ = transform_geometry(mesh_full, isobath_data, settings_dict, d_x, d_y)
+        if settings_dict["verbose"]:
+            # ---- Print alert
+            print(
+                """Longitude and latitude coordinates (WGS84) converted to standardized """
+                """coordinates (x and y)."""
+            )
+    else:
+        # ---- Else, duplicate the transect longitude and latitude coordinates as 'x' and 'y'
+        # -------- x
+        transect_data["x"] = transect_data["longitude"]
+        # -------- y
+        transect_data["y"] = transect_data["latitude"]
+        # ---- Duplicate the mesh grid longitude and latitude coordinates as 'x' and 'y'
+        # -------- x
+        mesh_full["x"] = mesh_full["longitude"]
+        # -------- y
+        mesh_full["y"] = mesh_full["latitude"]
+    # --- Append to the analysis attribute
+    analysis_dict.update({"kriging": {"mesh_df": mesh_full, "transect_df": transect_data}})
+
+    # Kriged results
+    kriged_results = kriging(
+        analysis_dict["kriging"]["transect_df"], analysis_dict["kriging"]["mesh_df"], settings_dict
+    )
+
+    # Stratified the kriging mesh
+    kriged_results["mesh_results_df"] = stratify_mesh(
+        input_dict, kriged_results["mesh_results_df"], settings_dict
+    )
+
+    # Return kriged (interpolated) results
+    return kriged_results, analysis_dict
+
+
+def apportion_kriged_values(
```

Comment:
This is a super long function -- I read through it at high level and think it is well done!
One suggestion: How about factoring out 1) the part related to imputation, and 2) the part about Additional reapportionment if age-1 fish are excluded? These seem to be fairly standalone, and if you factor them out, you can forget about the details when reading this function.